.github/workflows/main.yml

```diff
@@ -13,10 +14,22 @@ jobs:
     - run: npm ci && npm run build
     - run: npm test
 
+  access_check:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check user permissions
+      if: ${{ github.event_name == 'pull_request' && github.event.pull_request.author_association != 'MEMBER' }}
```

Comment:
The `pull_request_target` still uses the `event.pull_request` values from what I can tell: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target